sources/common/infoschema.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+type BaseInfoSchema interface {
+	GetBaseDdl() BaseToDdl
+	GetTableName(dbName string, tableName string) string
```

Comment:
I am wondering what's the reason for making this method public (similarly others below). Is it because Go compiler doesn't allow to call methods from concrete implementations otherwise, since implementations are in different packages?

Reply:
The compiler does not allow implementations in other packages to access private methods, so if we go with the approach of keeping a package per source, then we need to have public methods